olive/evaluator/metric.py

```diff
@@ -114,7 +140,7 @@ def validate_sub_types(cls, v, values):
                 raise ValueError(f"higher_is_better must be specified for ranked custom metric: {v['name']}")
             return v
         # name
-        sub_type_enum = AccuracySubType if values["type"] == MetricType.ACCURACY else LatencySubType
+        sub_type_enum = MetricType.map_to_metric_type(values["type"])
```

Comment:
why do we need convert from values["type"] to sub_type_enum? could not we use something like:
```python
if values["type"] == MetricType.ACCURACY:
    v["name"] = AccuracySubType(v["name"])
    metric_config_cls = HuggingfaceMetrics.get_config_class()
```

Reply:
Updated.